pl_examples/loop_examples/mnist_lite.py

```diff
@@ -0,0 +1,186 @@
+    def __init__(self, lite, args, model, dataloader):
+        super().__init__()
+        self.lite = lite
+        self.args = args
+        self.model = model
+        self.dataloader = dataloader
+        self.dataloader_iter = None
+        self.accuracy = Accuracy()
+        self.test_loss = 0
+
+    @property
+    def done(self) -> bool:
+        return False
+
+    def reset(self):
+        self.dataloader_iter = enumerate(self.dataloader)
+        self.test_loss = 0
+        self.accuracy.reset()
+
+    def advance(self) -> None:
+        _, (data, target) = next(self.dataloader_iter)
+        output = self.model(data)
+        self.test_loss += F.nll_loss(output, target)
+        self.accuracy(output, target)
+
+        if self.args.dry_run:
+            raise StopIteration
+
+    def on_run_end(self):
+        test_loss = self.lite.all_gather(self.test_loss).sum() / len(self.dataloader.dataset)
+
+        if self.lite.is_global_zero:
+            print(f"\nTest set: Average loss: {test_loss:.4f}, Accuracy: ({self.accuracy.compute():.0f}%)\n")
+
+
+class MainLoop(Loop):
+    def __init__(self, lite, args, model, optimizer, scheduler, train_loader, test_loader):
+        super().__init__()
+        self.lite = lite
+        self.args = args
+        self.epoch = 0
+        self.train_loop = TrainLoop(self.lite, self.args, model, optimizer, scheduler, train_loader)
+        self.test_loop = TestLoop(self.lite, self.args, model, test_loader)
+
+    @property
+    def done(self) -> bool:
+        return self.epoch >= self.args.epochs
+
+    def reset(self):
+        pass
+
+    def advance(self, *args: Any, **kwargs: Any) -> None:
+        self.train_loop.run(self.epoch)
+        self.test_loop.run()
+
+        if self.args.dry_run:
+            raise StopIteration
+
+        self.epoch += 1
+        self.lite.val_acc.reset()
+
+
+class Lite(LightningLite):
+    def run(self, hparams):
+        transform = T.Compose([T.ToTensor(), T.Normalize((0.1307,), (0.3081,))])
+        train_dataset = MNIST("./data", train=True, download=True, transform=transform)
+        test_dataset = MNIST("./data", train=False, transform=transform)
+        train_loader = torch.utils.data.DataLoader(train_dataset, hparams.batch_size)
+        test_loader = torch.utils.data.DataLoader(test_dataset, hparams.test_batch_size)
+
+        train_loader, test_loader = self.setup_dataloaders(train_loader, test_loader)
+
+        model = Net()
+        optimizer = optim.Adadelta(model.parameters(), lr=hparams.lr)
+
+        model, optimizer = self.setup(model, optimizer)
+        scheduler = StepLR(optimizer, step_size=1, gamma=hparams.gamma)
+
+        MainLoop(self, hparams, model, optimizer, scheduler, train_loader, test_loader).run()
+
+        if hparams.save_model:
+            self.save(model.state_dict(), "mnist_cnn.pt")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="LightningLite MNIST Example with Lightning Loops.")
```

Comment:
rather `LightningCLI`?

Reply:
LightningCLI is not compatible with Lite